### PR TITLE
Internal Solution for Clocktree (Closed PR#1470 and replaced with this) 

### PR DESCRIPTION
**Pull request by cskitty**
Added an explanation different from official solution (the final code is very similar though). 
Added video link to further explain the solution.

Let me know if the video is proper. I can create more videos later if it is okay.

@maggie-j-liu  I closed PR1470 as somehow my local git don't let me adding new files to this PR..

### CODE REVIEW

File: solutions/USACO feb20-clocktree-silver.mdx [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+Then Bessie visits another neighbor node, the sum of clocks of the second group increase by 1.
+As both groups increase clock by one, the difference between the sum of clocks for these two groups does not change.
+This process repeats until clocks in 1 or 2 groups reache all 12.
+
+We can check the initial sum difference between two groups (mod 12) to figure out the final clock numbers.
+
+1. If group0 is same as group2 (mod 12), Bessie can start from any nodes
+2. If group0 is 1 smaller than group1 (mod 12), Bessie starts from any node of group1
+3. if group1 is 1 smaller than group0 (mod 12), Bessie starts from any node of group0
+4. If group0 and group1 differ by more than 2 (mod 12), there's no way to get all 12
```
Comment (jessechoe10) [inline solutions/USACO feb20-clocktree-silver.mdx:29]: Doesn't the official solution already explain this well enough. Also, if you're going to put the video solution, you should remove this written solution.

Comment (maggie-j-liu) [reply solutions/USACO feb20-clocktree-silver.mdx:29]: > Doesn't the official solution already explain this well enough?

based on this; no https://github.com/cpinitiative/usaco-guide/issues/870#issue-816921172

> Also, if you're going to put the video solution, you should remove this written solution.

some people prefer text, others prefer video so i think it's fine

File: solutions/USACO feb20-clocktree-silver.mdx [hunk 2]
```diff
@@ -0,0 +1,109 @@
+---
+id: usaco-1016
+source: USACO Silver 2020 February
+title: Clock Tree
+author: Melody Yu
+---
+
+[Official Analysis](http://www.usaco.org/current/data/sol_clocktree_silver_feb20.html)
+
+We can build a graph with the rooms as nodes and corriddors as edges. Consider a simple case: there are 2 nodes in the graph and one edge.
```
Comment (nathangong) [inline solutions/USACO feb20-clocktree-silver.mdx:10]:
```suggestion
We can build a graph with the rooms as nodes and corridors as edges. Consider a simple case: there are 2 nodes in the graph and one edge.
```

File: solutions/USACO feb20-clocktree-silver.mdx [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+Bessie can start from one of the rooms, go to its neighbor, increase the clock go back and increase the starting room's clock.
```
Comment (nathangong) [inline solutions/USACO feb20-clocktree-silver.mdx:11]:
```suggestion
Bessie can start from one of the rooms, go to its neighbor, increase the clock, go back, and increase the starting room's clock.
```

File: solutions/USACO feb20-clocktree-silver.mdx [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+She can go back and forth until one (or both) rooms' clock reaches 12. During this process, we noticed that the difference between two rooms' clock readings are not changed.
+If the starting clock readings differ by one, Bessie can start from the bigger clock reading room and travels back and forth until both of the clocks point to 12.
```
Comment (nathangong) [inline solutions/USACO feb20-clocktree-silver.mdx:13]:
```suggestion
If the starting clock readings differ by one, Bessie can start from the bigger clock reading room and travel back and forth until both of the clocks point to 12.
```

File: solutions/USACO feb20-clocktree-silver.mdx [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+If the starting clock readings differ by more than one, it is impossible for Bessie to make both clocks point to 12.
+
+If there are more nodes, we can treat it similar to 2 nodes by partition all the nodes in to 2 groups and make this problem a biparte graph problem.
```
Comment (nathangong) [inline solutions/USACO feb20-clocktree-silver.mdx:16]:
```suggestion
If there are more nodes, we can treat it similar to 2 nodes by partitioning all the nodes into 2 groups and make this problem a biparte graph problem.
```

File: solutions/USACO feb20-clocktree-silver.mdx [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+We can partition the nodes into two groups using DFS if 2 neighbor nodes belongs to different groups.
```
Comment (nathangong) [inline solutions/USACO feb20-clocktree-silver.mdx:17]:
```suggestion
We can partition the nodes into two groups using DFS if 2 neighbor nodes belong to different groups.
```

File: solutions/USACO feb20-clocktree-silver.mdx [hunk 7]
```diff
@@ -0,0 +1,109 @@
...
+
+Every time Bessie visits a neighbor node, the sum of clocks of the group that Bessie visits increase by 1.
+Then Bessie visits another neighbor node, the sum of clocks of the second group increase by 1.
+As both groups increase clock by one, the difference between the sum of clocks for these two groups does not change.
+This process repeats until clocks in 1 or 2 groups reache all 12.
```
Comment (nathangong) [inline solutions/USACO feb20-clocktree-silver.mdx:22]:
```suggestion
This process repeats until clocks in 1 or 2 groups reach all 12.
```

**Review by nathangong — CHANGES_REQUESTED** [review]
I found a few typos that you might want to fix.

**Review by frodakcin — CHANGES_REQUESTED** [review]
Video looks great; well done! :)

Explanation is good, save a few typos & optional improvements.
1. It might help to have more use of latex (for numbers, variable names like "group1," etc)
2. It might help to prove/vaguely explain existence of solution given condition is satisfied

File: solutions/USACO feb20-clocktree-silver.mdx [hunk 8]
```diff
@@ -0,0 +1,109 @@
...
+[Official Analysis](http://www.usaco.org/current/data/sol_clocktree_silver_feb20.html)
+
+We can build a graph with the rooms as nodes and corridors as edges. Consider a simple case: there are 2 nodes in the graph and one edge.
+Bessie can start from one of the rooms, go to its neighbor, increase the clock, go back, and increase the starting room's clock.
+She can go back and forth until one (or both) rooms' clock reaches 12. During this process, we noticed that the difference between two rooms' clock readings are not changed.
+If the starting clock readings differ by one, Bessie can start from the bigger clock reading room and travel back and forth until both of the clocks point to 12.
+If the starting clock readings differ by more than one, it is impossible for Bessie to make both clocks point to 12.
+
+If there are more nodes, we can treat it similar to 2 nodes by partitioning all the nodes into 2 groups and make this problem a biparte graph problem.
+We can partition the nodes into two groups using DFS if 2 neighbor nodes belong to different groups.
```
[the remark this answers is not in the document]
Comment (cskitty) [reply solutions/USACO feb20-clocktree-silver.mdx:17]: Sorry, what do you mean by "all trees are bipartite" ? Can you explain a bit further where :.D?

[the remark this answers is not in the document]
Comment (frodakcin) [reply solutions/USACO feb20-clocktree-silver.mdx:17]: All trees are bipartite graphs. (this two coloring will never run into coloring a node in multi colors)

File: solutions/USACO feb20-clocktree-silver.mdx [hunk 9]
```diff
@@ -0,0 +1,108 @@
...
+
+[Official Analysis](http://www.usaco.org/current/data/sol_clocktree_silver_feb20.html)
+
+We can build a graph with the rooms as nodes and corridors as edges. Consider a simple case: there are 2 nodes in the graph and one edge.
+Bessie can start from one of the rooms, go to its neighbor, increase the clock, go back, and increase the starting room's clock.
+During this process of moving back and forth, notice that the difference between two rooms' clock readings may only change by 1.
+If the starting clock readings differ by one, Bessie can start from the bigger clock reading room and travel back and forth until both of the clocks point to 12.
+If the starting clock readings differ by more than one, it is impossible for Bessie to make both clocks point to 12.
+
+If there are more nodes, we can treat it similar to 2 nodes by partitioning all the nodes into 2 groups and making this problem a bipartite graph problem. All trees here are biparte graphs, meaning no nodes will have multiple colors. We can partition the nodes into two groups using DFS, assuming that any two neighbor nodes belong to different groups.
```
Comment (frodakcin) [inline solutions/USACO feb20-clocktree-silver.mdx:16]:
```suggestion
If there are more nodes, we can treat it similar to two node case by considering the two groups in a bipartite partition. Note that all trees are bipartite graphs, so this partition will always be successful.
```

Just a heads up, it's "bipartite," not "biparte."

**Review by frodakcin — CHANGES_REQUESTED** [review]
probably better wording? Now that i think about it there's video so the written expl doesnt need to be perfect.

Also, latex would make it read smoother but its probably not necessary

File: solutions/USACO feb20-clocktree-silver.mdx (hunk 9 above)
Comment (cskitty) [reply solutions/USACO feb20-clocktree-silver.mdx]: @frodakcin committed, thanks! Could you check it again?

File: solutions/USACO feb20-clocktree-silver.mdx [hunk 10]
```diff
@@ -0,0 +1,108 @@
...
+
+## Video Solution
+
+<Youtube id="vMA54A2AT0I" />
+
+<LanguageSection>
+
+<CPPSection>
+
+```cpp
```
Comment (maggie-j-liu) [inline solutions/USACO feb20-clocktree-silver.mdx:38]: use tabs instead of spaces to indent

File: solutions/USACO feb20-clocktree-silver.mdx [hunk 11]
```diff
@@ -0,0 +1,108 @@
...
+    // update color/sum
+    if (color == 0) {
+        nodes0++;
+        sum0 += A[i];
+    } else {
+        nodes1++;
+        sum1 += A[i];
+    }
+
+    for(int j=0;j<edges[i].size();j++) {
```
Comment (maggie-j-liu) [inline solutions/USACO feb20-clocktree-silver.mdx:57]:
```suggestion
    for(int j = 0; j < edges[i].size(); j++) {
```

File: solutions/USACO feb20-clocktree-silver.mdx [hunk 12]
```diff
@@ -0,0 +1,108 @@
...
+    }
+
+    for(int i = 1; i < N; i++) {
+        cin >> a >> b;
+        a--,b--;
+        edges[a].push_back(b);
+        edges[b].push_back(a);
+    }
+
+    dfs(0,0,-1);
```
Comment (maggie-j-liu) [inline solutions/USACO feb20-clocktree-silver.mdx:83]:
```suggestion
    freopen("clocktree.in", "r", stdin);
    freopen("clocktree.out", "w", stdout);

    int a, b;
    cin >> N;

    for(int i = 0; i < N; i++) {
        cin >> A[i];
    }

    for(int i = 1; i < N; i++) {
        cin >> a >> b;
        a--, b--;
        edges[a].push_back(b);
        edges[b].push_back(a);
    }

    dfs(0, 0, -1);
```

File: solutions/USACO feb20-clocktree-silver.mdx [hunk 13]
```diff
@@ -0,0 +1,108 @@
...
+    // same mod means you can start anywhere
+    if((sum0 % 12) == (sum1 % 12)) {
+        cout << N;
+    }
+        // if group0 is 1 smaller than group1, start from group1
+    else if ((sum0 + 1)%12 == (sum1 % 12)) {
+        cout << nodes1;
+    }
+        // if group1 is 1 smaller than group0, start from group0
+    else if((sum0 % 12) == ((sum1 + 1)%12)) {
```
Comment (maggie-j-liu) [inline solutions/USACO feb20-clocktree-silver.mdx:94]:
```suggestion
    else if((sum0 + 1) % 12 == (sum1 % 12)) {
        cout << nodes1;
    }
        // if group1 is 1 smaller than group0, start from group0
    else if((sum0 % 12) == ((sum1 + 1) % 12)) {
```

File: solutions/USACO feb20-clocktree-silver.mdx [hunk 14]
```diff
@@ -0,0 +1,108 @@
...
+        edges[b].push_back(a);
+    }
+
+    dfs(0,0,-1);
+
+    // same mod means you can start anywhere
+    if((sum0 % 12) == (sum1 % 12)) {
+        cout << N;
+    }
+        // if group0 is 1 smaller than group1, start from group1
```
Comment (maggie-j-liu) [inline solutions/USACO feb20-clocktree-silver.mdx:89]: are these comments supposed to be indented?

**Review by maggie-j-liu — CHANGES_REQUESTED** [review]

File: solutions/USACO feb20-clocktree-silver.mdx (hunk 14 above)
Comment (cskitty) [reply solutions/USACO feb20-clocktree-silver.mdx]: Edited, thanks!

File: solutions/USACO feb20-clocktree-silver.mdx (hunk 10 above)
Comment (cskitty) [reply solutions/USACO feb20-clocktree-silver.mdx:38]: Edited, thanks!

Comment (maggie-j-liu) [reply solutions/USACO feb20-clocktree-silver.mdx:38]: it still looks like spaces to me?

Comment (cskitty) [reply solutions/USACO feb20-clocktree-silver.mdx:38]: I tried changing the spaces to tabs, is it fine now? By the way how do you visually check spaces vs tabs in github? I really don't know how to X,D

**Review by frodakcin — APPROVED** [review]

File: solutions/USACO feb20-clocktree-silver.mdx (hunk 10 above)
Comment (maggie-j-liu) [reply solutions/USACO feb20-clocktree-silver.mdx:38]: still looks like spaces to me... on github, tabs are wider than 4 spaces (and in editor it looks different)
spaces: 
image
tabs:
image

File: solutions/USACO feb20-clocktree-silver.mdx [hunk 15]
```diff
@@ -0,0 +1,109 @@
...
+If there are more nodes, we can treat it similar to two node case by considering the two groups in a bipartite partition. Note that all trees are bipartite graphs, so this partition will always be successful.
+
+Every time Bessie visits a neighbor node, the sum of clocks of the group that Bessie visits increase by 1.
+Then Bessie visits another neighbor node, the sum of clocks of the second group increase by 1.
+As both groups increase clock by one, the difference between the sum of clocks for these two groups does not change.
+This process repeats until clocks in 1 or 2 groups reach all 12.
+
+We can check the initial sum difference between two groups (mod 12) to figure out the final clock numbers.
+
+1. If group0 is same as group2 (mod 12), Bessie can start from any nodes
```
Comment (maggie-j-liu) [inline solutions/USACO feb20-clocktree-silver.mdx:25]:
```suggestion
1. If group0 is same as group2 (mod 12), Bessie can start from any node
```

**Review by maggie-j-liu — APPROVED** [review]

**Review by nathangong — APPROVED** [review]

File: solutions/USACO feb20-clocktree-silver.mdx [hunk 16]
```diff
@@ -0,0 +1,109 @@
...
+			dfs(edges[i][j], 1 - color, i);
+		}
+	}
+}
+
+int main() {
+	freopen("clocktree.in", "r", stdin);
+	freopen("clocktree.out", "w", stdout);
+	
+	int a, b;
```
Comment (bqi343) [inline solutions/USACO feb20-clocktree-silver.mdx:70]: why are `a,b` up here

**Review by bqi343 — CHANGES_REQUESTED** [review]

File: solutions/USACO feb20-clocktree-silver.mdx (hunk 16 above)
Comment (cskitty) [reply solutions/USACO feb20-clocktree-silver.mdx]: Edited and moved a, b to inside the loop

File: solutions/USACO feb20-clocktree-silver.mdx [hunk 17]
```diff
@@ -0,0 +1,109 @@
...
+	if (color == 0) {
+		nodes0++;
+		sum0 += A[i];
+	}
+	else {
+		nodes1++;
+		sum1 += A[i];
+	}
+	
+	for(int j = 0; j < edges[i].size(); j++) {
```
Comment (bqi343) [inline solutions/USACO feb20-clocktree-silver.mdx:58]: also pretty minor but why don't you use a for-each loop here? (`for (int j: edges[i])`)

**Review by bqi343 — CHANGES_REQUESTED** [review]

File: solutions/USACO feb20-clocktree-silver.mdx [hunk 18]
```diff
@@ -0,0 +1,109 @@
...
+
+We can build a graph with the rooms as nodes and corridors as edges. Consider a simple case: there are 2 nodes in the graph and one edge.
+Bessie can start from one of the rooms, go to its neighbor, increase the clock, go back, and increase the starting room's clock.
+During this process of moving back and forth, notice that the difference between two rooms' clock readings may only change by 1.
+If the starting clock readings differ by one, Bessie can start from the bigger clock reading room and travel back and forth until both of the clocks point to 12.
+If the starting clock readings differ by more than one, it is impossible for Bessie to make both clocks point to 12.
+
+If there are more nodes, we can treat it similar to two node case by considering the two groups in a bipartite partition. Note that all trees are bipartite graphs, so this partition will always be successful.
+
+Every time Bessie visits a neighbor node, the sum of clocks of the group that Bessie visits increase by 1.
```
Comment (bqi343) [inline solutions/USACO feb20-clocktree-silver.mdx:18]:
```suggestion
Every time Bessie visits a neighbor node, the sum of clocks of the group that Bessie visits increases by 1.
```

File: solutions/USACO feb20-clocktree-silver.mdx [hunk 19]
```diff
@@ -0,0 +1,109 @@
...
+Then Bessie visits another neighbor node, the sum of clocks of the second group increase by 1.
```
Comment (bqi343) [inline solutions/USACO feb20-clocktree-silver.mdx:19]:
```suggestion
Then Bessie visits another neighbor node, the sum of clocks of the second group increases by 1.
```

File: solutions/USACO feb20-clocktree-silver.mdx [hunk 20]
```diff
@@ -0,0 +1,109 @@
...
+As both groups increase clock by one, the difference between the sum of clocks for these two groups does not change.
```
Comment (bqi343) [inline solutions/USACO feb20-clocktree-silver.mdx:20]: "increase clock by one" sounds odd to me

```suggestion
As both sums increase by one, the difference between the sums does not change.
```

File: solutions/USACO feb20-clocktree-silver.mdx [hunk 21]
```diff
@@ -0,0 +1,109 @@
...
+This process repeats until clocks in 1 or 2 groups reach all 12.
```
Comment (bqi343) [inline solutions/USACO feb20-clocktree-silver.mdx:21]:
```suggestion
This process repeats until all clocks reach 12.
```

File: solutions/USACO feb20-clocktree-silver.mdx [hunk 22]
```diff
@@ -0,0 +1,109 @@
...
+
+We can check the initial sum difference between two groups (mod 12) to figure out the final clock numbers.
```
Comment (bqi343) [inline solutions/USACO feb20-clocktree-silver.mdx:23]:
```suggestion
We can check the initial sum difference between the two groups (mod 12) to figure out the final clock numbers.
```

File: solutions/USACO feb20-clocktree-silver.mdx [hunk 23]
```diff
@@ -0,0 +1,109 @@
...
+
+1. If group0 is same as group2 (mod 12), Bessie can start from any node
+2. If group0 is 1 smaller than group1 (mod 12), Bessie starts from any node of group1
+3. if group1 is 1 smaller than group0 (mod 12), Bessie starts from any node of group0
```
Comment (bqi343) [inline solutions/USACO feb20-clocktree-silver.mdx:27]:
```suggestion
3. If group1 is 1 smaller than group0 (mod 12), Bessie starts from any node of group0
```

File: solutions/USACO feb20-clocktree-silver.mdx [hunk 24]
```diff
@@ -0,0 +1,109 @@
...
+If there are more nodes, we can treat it similar to two node case by considering the two groups in a bipartite partition. Note that all trees are bipartite graphs, so this partition will always be successful.
+
+Every time Bessie visits a neighbor node, the sum of clocks of the group that Bessie visits increase by 1.
+Then Bessie visits another neighbor node, the sum of clocks of the second group increase by 1.
+As both groups increase clock by one, the difference between the sum of clocks for these two groups does not change.
+This process repeats until clocks in 1 or 2 groups reach all 12.
+
+We can check the initial sum difference between two groups (mod 12) to figure out the final clock numbers.
+
+1. If group0 is same as group2 (mod 12), Bessie can start from any node
```
Comment (bqi343) [inline solutions/USACO feb20-clocktree-silver.mdx:25]:
```suggestion
1. If group0 is the same as group1 (mod 12), Bessie can start from any node.
```

Comment (bqi343) [reply solutions/USACO feb20-clocktree-silver.mdx:25]: also, group and group1 were never defined

Comment (bqi343) [reply solutions/USACO feb20-clocktree-silver.mdx:25]: other lines should end with periods as well

**Review by bqi343 — CHANGES_REQUESTED** [review]

File: solutions/USACO feb20-clocktree-silver.mdx [hunk 25]
```diff
@@ -0,0 +1,109 @@
...
+If there are more nodes, we can treat it similar to the two node case by considering the two groups in a bipartite partition. As all trees are bipartite graphs, this partition always exists.
+
+The first time Bessie visits a neighbor node, the sum of clocks of the first group increases by 1.
+Then Bessie visits another neighbor node, the sum of clocks of the second group increases by 1.
+As both sums increase by one, the difference between the sums does not change.
+This process repeats until all clocks reach 12.
+
+We can check the initial sum difference between the two groups (mod 12) to figure out the final clock numbers.
+
+1. If group0 is the same as group1 (mod 12), Bessie can start from any node.
```
Comment (bqi343) [inline solutions/USACO feb20-clocktree-silver.mdx:25]: The last two comments here were not resolved:

https://github.com/cpinitiative/usaco-guide/pull/1474#discussion_r637164340

File: solutions/USACO feb20-clocktree-silver.mdx [hunk 26]
```diff
@@ -0,0 +1,109 @@
...
+2. If group0 is 1 smaller than group1 (mod 12), Bessie starts from any node of group1
+3. If group1 is 1 smaller than group0 (mod 12), Bessie starts from any node of group0
+4. If group0 and group1 differ by more than 2 (mod 12), there's no way to get all 12
```
Comment (bqi343) [inline solutions/USACO feb20-clocktree-silver.mdx:28]:
```suggestion
4. If group0 and group1 differ by at least 2 (mod 12), there's no way to get all 12
```

**Review by bqi343 — CHANGES_REQUESTED** [review]

File: solutions/USACO feb20-clocktree-silver.mdx (hunk 25 above)
Comment (thecodingwizard) [reply solutions/USACO feb20-clocktree-silver.mdx:25]: oops, missed this D: @cskitty ?

File: solutions/USACO feb20-clocktree-silver.mdx (hunk 24 above)
Comment (cskitty) [reply solutions/USACO feb20-clocktree-silver.mdx]: Fixed both, sorry!

File: solutions/USACO feb20-clocktree-silver.mdx (hunk 25 above)
Comment (cskitty) [reply solutions/USACO feb20-clocktree-silver.mdx:25]: Fixed the comments, sorry about the lateness!

File: solutions/USACO feb20-clocktree-silver.mdx [hunk 27]
```diff
@@ -0,0 +1,109 @@
...
+If the starting clock readings differ by more than one, it is impossible for Bessie to make both clocks point to 12.
+
+If there are more nodes, we can treat it similar to the two node case by considering the two groups in a bipartite partition. As all trees are bipartite graphs, this partition always exists.
+
+The first time Bessie visits a neighbor node, the sum of clocks of the first group increases by 1.
+Then Bessie visits another neighbor node, the sum of clocks of the second group increases by 1.
+As both sums increase by one, the difference between the sums does not change.
+This process repeats until all clocks reach 12.
+
+We can check the initial sum difference between the two groups (mod 12) to figure out the final clock numbers, where we define group0 and group1 as the two remaining groups we have.
```
Comment (bqi343) [inline solutions/USACO feb20-clocktree-silver.mdx:23]: > the two remaining groups we have

what does "remaining" mean here?

**Review by bqi343 — CHANGES_REQUESTED** [review]

File: solutions/USACO feb20-clocktree-silver.mdx (hunk 27 above)
Comment (cskitty) [reply solutions/USACO feb20-clocktree-silver.mdx]: Is it clearer now?

File: solutions/USACO feb20-clocktree-silver.mdx [hunk 28]
```diff
@@ -20,7 +20,7 @@ Then Bessie visits another neighbor node, the sum of clocks of the second group
 As both sums increase by one, the difference between the sums does not change.
 This process repeats until all clocks reach 12.
 
-We can check the initial sum difference between the two groups (mod 12) to figure out the final clock numbers, where we define group0 and group1 as the two remaining groups we have.
+We can check the initial sum difference between the two groups (mod 12) to figure out the final clock numbers, where we define group0 and group1 as the two groups we get after putting all rooms into the biparte partition.
```
Comment (bqi343) [inline solutions/USACO feb20-clocktree-silver.mdx:23]: bipartite

**Review by bqi343 — CHANGES_REQUESTED** [review]

File: solutions/USACO feb20-clocktree-silver.mdx (hunk 28 above)
Comment (cskitty) [reply solutions/USACO feb20-clocktree-silver.mdx]: sorry, changed!

File: solutions/USACO feb20-clocktree-silver.mdx [hunk 29]
```diff
@@ -0,0 +1,109 @@
...
+If the starting clock readings differ by more than one, it is impossible for Bessie to make both clocks point to 12.
+
+If there are more nodes, we can treat it similar to the two node case by considering the two groups in a bipartite partition. As all trees are bipartite graphs, this partition always exists.
+
+The first time Bessie visits a neighbor node, the sum of clocks of the first group increases by 1.
+Then Bessie visits another neighbor node, the sum of clocks of the second group increases by 1.
+As both sums increase by one, the difference between the sums does not change.
+This process repeats until all clocks reach 12.
+
+We can check the initial sum difference between the two groups (mod 12) to figure out the final clock numbers, where we define group0 and group1 as the two groups we get after putting all rooms into the bipartite partition.
```
Comment (bqi343) [inline solutions/USACO feb20-clocktree-silver.mdx:23]: aren't group0 and group1 actually defined to be the sums of the clocks in each of the two groups of the bipartite partition, not the groups themselves?

**Review by bqi343 — CHANGES_REQUESTED** [review]

File: solutions/USACO feb20-clocktree-silver.mdx (hunk 29 above)
Comment (cskitty) [reply solutions/USACO feb20-clocktree-silver.mdx]: How is it now?

Comment (bqi343) [reply solutions/USACO feb20-clocktree-silver.mdx]: looks good, thanks!

**Review by bqi343 — APPROVED** [review]

**Pull request merged**